benchopt/utils/temp_benchmark.py

```diff
@@ -0,0 +1,51 @@
+import os
+import tempfile
+import contextlib
+
+from pathlib import Path
+
+from benchopt.benchmark import Benchmark
+from benchopt.tests import DUMMY_BENCHMARK_PATH
+
+
+dummy_objective = (DUMMY_BENCHMARK_PATH / 'objective.py').read_text()
+
+dummy_solvers = [
+    (DUMMY_BENCHMARK_PATH / "solvers" / p).read_text()
+    for p in os.listdir(DUMMY_BENCHMARK_PATH / "solvers")
+    if p.endswith(".py")]
+
+dummy_datasets = [
+    (DUMMY_BENCHMARK_PATH / "datasets" / p).read_text()
+    for p in os.listdir(DUMMY_BENCHMARK_PATH / "datasets")
+    if p.endswith(".py")]
+
+
+@contextlib.contextmanager
+def temp_benchmark(
+        objective=dummy_objective, datasets=None,
+        solvers=None, config=None):
+    if solvers is None:
+        solvers = dummy_solvers
+    if datasets is None:
+        datasets = dummy_datasets
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        temp_path = Path(tempdir)
+        os.mkdir(temp_path / "solvers")
+        os.mkdir(temp_path / "datasets")
```

Comment:
We can just use the `Pathlib` API
```suggestion
        (temp_path / "solvers").mkdir()
        (temp_path / "datasets").mkdir()
```